module/os/linux/zfs/zpl_ctldir.c

```diff
@@ -248,7 +248,7 @@ zpl_snapdir_iterate(struct file *filp, zpl_dir_context_t *ctx)
 	if (!zpl_dir_emit_dots(filp, ctx))
 		goto out;
 
-	pos = ctx->pos;
+	pos = (ctx->pos == 2 ? 0 : ctx->pos);
 	while (error == 0) {
```

Comment:
this test looks so weird that it might be worth a comment...

Reply:
Thanks ceedriic.